### ISSUE TITLE
About the generated dataset size

### ISSUE THREAD

**Issue by pjl013** [hxwork/HandBooster_Pytorch#5]
Thank you for your extremely exciting work. May I ask about the size of the generated dataset that you ultimately used for training?

**Comment by hxwork** [comment]
Hi,

We generate the same size as the original training data (DexYCB, HO3D).

**Comment by pjl013** [comment]
Thanks  for your early response, 
but I still have a question about this part. In your paper, you mentioned that you used the Dexgraspnet method to 'perform this simulation 10,000 times on each object,' then 'align the orientation of the generated pose to a random one from the training set,' and finally perform filtering and other steps. However, even without considering the significant reduction in samples due to filtering, the scale of the dataset (DexYCB) you ultimately constructed seems much larger than 10,000 samples per object. Could you kindly clarify whether you finally generated multiple data samples from a single grasp simulation result?

**Comment by hxwork** [comment]
Actually, the same pose can correspond to several different camera views, e.g., the DexYCB dataset (s0 split) has 8 views for each hand pose. So if you generate `n` poses, you can align their global orientations to `m` views and then generate `n*m` RGB images.

Also, the final data size can be much larger than the original number of the poses since we apply rotation augmentations to the poses after alignment.

**Comment by pjl013** [comment]
Thank you for your response. Does the 'n' mentioned refer to 10,000 as stated in the paper? Also, is 'm' selected to be 8, as in the Dex-YCB dataset?Besides, could you provide more details on the specific implementation of the rotation augmentations?Finally, considering the different viewpoints and rotation augmentations, how many 3D grasp poses of objects were generated in total?

**Comment by hxwork** [comment]
Yes, but for the poses used in the image generation, we only selected the validated poses, i.e., "no obvious penetration between the hand and the object" (by calculating the volume of the through-mold part and treating poses as invalid if their volumes are larger than a pre-defined threshold) and "the hand has to contact the object" (the DexGraspNet may fail to obtain reasonable results when the object is too small, e.g., marker). Also, we removed one of the original constraints (related to the initial object pose on the table) in the DexGraspNet when simulating the grasping poses. The number of 3D grasping poses should be less than 10,000, which depends on how many of them are valid.

The implementation of the rotation augmentation is quite simple, we just added random 3D rotations within a pre-defined range ([-30, 30] degrees along XYZ axes) on the simulated poses after alignment.

**Comment by pjl013** [comment]
Thank you for your response. Could you please provide the pre-defined threshold you mentioned during the selection process?

**Comment by hxwork** [comment]
Hi,

Sorry for the late reply, we remove the grasping poses with self-intersection or cross-intersection between hands, and the pre-defined threshold is set to 5e-7 (larger than this threshold should be invalid). We use pymeshlab to calculate the mesh volume.

**Comment by pjl013** [comment]
Thank you for sharing your work. I have completed the preliminary selection of hand models and obtained the 3D grasping models for generating images. What confuses me is that even though I align my synthetic models using multiple poses before selection, and then apply multiple rotation augmentations to gain a large number of samples, I still can't obtain enough results after the selection to generate data on the scale of datasets like dex_ycb for subsequent use. Could you tell me if can I still apply multi-view and rotation augmentations to increase the data volume after the selection?

**Comment by hxwork** [comment]
I am not sure about your selection part. Why you can't obtain enough data after the selection? Have you checked if your selection process works as you expected, e.g., visualizing some cases?

**Comment by pjl013** [comment]
I have solved the issue, thank you for your response. The problem arose because normalized probabilities were used for filtering during the selection process, so simply doubling the size of dataset before filtering does not lead to a similar doubling of the dataset after filtering. Now, by using a much larger size of dataset than what is ultimately needed, I was able to obtain the desired amount of data, and the visualization results are in line with my expectations. Once again, thank you for all the help you've provided throughout, as well as for open-sourcing the project.

**Comment by hxwork** [comment]
Thanks, then I will close this issue.
